### DISCUSSION TITLE
What happens to large number of queries stuck in prisma_client_queries_wait?

### DISCUSSION THREAD

**Discussion by tomcru** [prisma/prisma#21584] [Q&A]
### Question

Executing a large number of queries with `?pool_timeout=2` (in this case) has a lot of queries landing in `prisma_client_queries_wait`. Why are these queries waiting forever? I'm expecting that they would be removed from the wait after 2s and fail - but they seem to be stuck inside `prisma_client_queries_wait` forever (the number stays the same - even with no new incoming queries).

example prisma metrics:
```
{
  "counters": [
    {
      "key": "prisma_client_queries_total",
      "labels": {},
      "value": 71720,
      "description": "The total number of Prisma Client queries executed"
    },
    {
      "key": "prisma_datasource_queries_total",
      "labels": {},
      "value": 71850,
      "description": "The total number of datasource queries executed"
    },
    {
      "key": "prisma_pool_connections_closed_total",
      "labels": {},
      "value": 0,
      "description": "The total number of pool connections closed"
    },
    {
      "key": "prisma_pool_connections_opened_total",
      "labels": {},
      "value": 40,
      "description": "The total number of pool connections opened"
    }
  ],
  "gauges": [
    {
      "key": "prisma_client_queries_active",
      "labels": {},
      "value": 0,
      "description": "The number of currently active Prisma Client queries"
    },
    {
      "key": "prisma_client_queries_wait",
      "labels": {},
      "value": 156304,
      "description": "The number of datasource queries currently waiting for an free connection"
    },
    {
      "key": "prisma_pool_connections_busy",
      "labels": {},
      "value": 0,
      "description": "The number of pool connections currently executing datasource queries"
    },
    {
      "key": "prisma_pool_connections_idle",
      "labels": {},
      "value": 40,
      "description": "The number of pool connections that are not busy running a query"
    },
    {
      "key": "prisma_pool_connections_open",
      "labels": {},
      "value": 40,
      "description": "The number of pool connections currently open"
    }
  ],
}
```

### Expected behavior (optional)

queries are removed from waiting when they take longer than ?pool_timeout to find a connection.

### Information about Prisma Schema, Client Queries and Environment (optional)

- Database: PostgreSQL
- Node.js version: v20.5.0

- prisma: 5.4.2
- @prisma/client : 5.4.2

**Comment by janpio** [comment]
Do the queries fail?
If so, then this might be a bug that on failure the "waiting" queries are not removed from the metrics.
(As `metrics` are a preview feature, problems like this are expected and exactly why we are doing the preview feature phase actually)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
